src/h5web/vis-packs/nexus/containers/NxImageContainer.tsx

```diff
@@ -1,58 +1,47 @@
-import { ReactElement, useEffect } from 'react';
-import { assertArray, assertGroup, assertOptionalStr } from '../../../guards';
+import { ReactElement, Suspense } from 'react';
+import { assertGroup, assertMinDims } from '../../../guards';
 import type { VisContainerProps } from '../../models';
 import MappedHeatmapVis from '../../core/heatmap/MappedHeatmapVis';
-import { useHeatmapConfig } from '../../core/heatmap/config';
 import { useAxisMapping, useNxData } from '../hooks';
 import { useDatasetValue } from '../../core/hooks';
 import { getDatasetLabel } from '../utils';
 import { useDimMappingState } from '../../hooks';
 import DimensionMapper from '../../../dimension-mapper/DimensionMapper';
+import ValueLoader from '../../../visualizer/ValueLoader';
 
 function NxImageContainer(props: VisContainerProps): ReactElement {
   const { entity } = props;
   assertGroup(entity);
 
   const nxData = useNxData(entity);
+
   const { signalDataset, titleDataset, axisDatasetMapping, silxStyle } = nxData;
   const { axesScaleType, signalScaleType } = silxStyle;
+  assertMinDims(signalDataset, 2);
 
   const { dims } = signalDataset.shape;
-  if (dims.length < 2) {
-    throw new Error('Expected signal dataset with at least two dimensions');
-  }
-
   const [dimMapping, setDimMapping] = useDimMappingState(dims, 2);
 
-  const value = useDatasetValue(signalDataset);
-  assertArray<number>(value);
-
   const title = useDatasetValue(titleDataset);
-  assertOptionalStr(title);
-
   const axisMapping = useAxisMapping(axisDatasetMapping, axesScaleType);
 
-  const { setScaleType } = useHeatmapConfig();
-  useEffect(() => {
-    if (signalScaleType) {
-      setScaleType(signalScaleType);
-    }
-  }, [setScaleType, signalScaleType]);
-
   return (
     <>
       <DimensionMapper
         rawDims={dims}
         mapperState={dimMapping}
         onChange={setDimMapping}
       />
-      <MappedHeatmapVis
-        value={value}
-        dims={dims}
-        dimMapping={dimMapping}
-        axisMapping={axisMapping}
-        title={title || getDatasetLabel(signalDataset)}
-      />
+      <Suspense fallback={<ValueLoader />}>
+        <MappedHeatmapVis
+          dataset={signalDataset}
+          dims={dims}
```

Comment:
It won't be relevant to pass `dims` later on, since we'll always have a 2D slice to pass to `useBaseArray`.